core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -1643,7 +1643,15 @@ components:
         ledger_hashes:
           $ref: "#/components/schemas/LedgerHashes"
         proposer_timestamp:
-          $ref: '#/components/schemas/Instant'
+          description: |
+            The time at which the consensus leader created the proposal for extending the ledger to
+            the represented point.
+
+            Note: in abnormal cases (e.g. Byzantine network quorum), this on-ledger field may be set
```

Comment:
> it's technically an invalid schema

I thought that it actually IS technically valid, but all the siblings of `$ref` are technically ignored - that's how I understand the [$ref and Sibling Elements](https://swagger.io/docs/specification/using-ref/).

OFC in practice we rely on our doc generator NOT ignoring the siblings:

<img width="844" alt="image" src="https://github.com/radixdlt/babylon-node/assets/119413677/e648d113-38c1-4a84-9b19-309aa2a15f61">

_(a correctly rendered `description` next to `$ref: Instant` [here](https://radix-babylon-core-api.redoc.ly/#tag/State/paths/~1state~1consensus-manager/post!c=200&path=current_time_rounded_to_minutes/31/value/proposer_timestamp_rounded_down_to_minute&t=response))_

> we just put this Note: on `InstantMs.date_time` and `InstantMs. unix_timestamp_ms` instead

Unfortunately there are some instances of `InstantMs` for which this note would be a lie, e.g. `retry_from_timestamp` (there is no way the Node can output >10^14 there, and also the value does not come from the ledger). This description really belongs to the field, not the type.

Additionally, we want the `description`s on our instant fields for other reasons - namely for describing what each instant means (not everything is obvious from a field name).

My current position is that putting `description` on a field is a practical win:
- our generator correctly renders the field-specific descriptions for our public docs,
- other people's generators correctly generate types' sources for them, because the schema is technically valid (worst case - some descriptions will be ignored and missing in those sources)

Reply:
Okay, I'm comfortable with that